### DISCUSSION TITLE
<sanity check>Handling reconnects of websocket on dodgy network link

### DISCUSSION THREAD

**Discussion by yatesco** [tokio-rs/axum#612] [Q&A] [answered]
Hi there, do web sockets have any notion of "the same client reconnected" or are they distinct "anonymous" connections? 

I assume so, so I am sticking any client specific state into a session, but I wanted to check. In the chat example, If "Bob" disconnects and reconnects because the network dropped out then I assume he is an entirely different "Bob".

Thanks!

**Comment by davidpdrsn** [comment, marked as answer]
Not that I'm aware of. Clients have to reconnect and get authorized somehow. How to do that is up to you and not WebSocket specific.

**Reply by yatesco** [reply]
thanks David.
